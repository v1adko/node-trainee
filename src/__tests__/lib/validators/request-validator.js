import Joi from 'joi';
import requestValidator from '../../../lib/validators/request-validator';

const testSchemaValidation = Joi.string()
  .alphanum()
  .required();

const methodValidationRules = {
  testField: testSchemaValidation
};

const request = {
  data: { testField: 'testTextForField' }
};
describe('Test request validator', async () => {
  it('should request field and pass it, because they are valid', async () => {
    await requestValidator(methodValidationRules, request);
  });

  it('should pass all requests, because rules for method does not exist ', async () => {
    await requestValidator(null, request);
  });

  it('should return RequestValidationError, because request data contain non alphanum characters', async () => {
    const invalidRequest = {
      data: { testField: 'test text for field' }
    };
    expect(
      requestValidator(methodValidationRules, invalidRequest)
    ).rejects.toMatchSnapshot();
  });

  it('should return RequestValidationError, because request data contain non alphanum characters 2', async () => {
    const data = { testField: 'testTextForField' };
    Object.defineProperty(data, 'testField', {
      get() {
        throw new Error('test error');
      }
    });
    const thowingRequest = { data };

    expect(
      requestValidator(methodValidationRules, thowingRequest)
    ).rejects.toMatchSnapshot();
  });
});
