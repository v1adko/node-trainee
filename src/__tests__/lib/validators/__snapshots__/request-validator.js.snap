// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test request validator should return RequestValidationError, because request data contain non alphanum characters 1`] = `[RequestValidationError: "testField" must only contain alpha-numeric characters]`;

exports[`Test request validator should return RequestValidationError, because request data contain non alphanum characters 2 1`] = `[Error: test error]`;
